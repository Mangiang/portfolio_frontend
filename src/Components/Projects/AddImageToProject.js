import React, {Component} from 'react';
import {hot} from 'react-hot-loader';

import {connect} from 'react-redux';
import {uploadImages} from '../../Actions/Actions';


class AddImageToProject extends Component {
    constructor(props) {
        super(props);
        this.state = {
            images: []
        };

        this.onClickUpdate = this.onClickUpdate.bind(this);
        this.updateImages = this.updateImages.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.state.token = nextProps.token;
        this.setState({
            images: []
        })
    }

    componentWillMount() {
        this.state.token = this.props.token;
    }

    onClickUpdate() {
        this.props.dispatch(uploadImages(
            this.state.images,
            this.props.id,
            this.state.token
        ));
        this.setState({
            images: []
        })
    }

    updateImages(imgList) {
        let images = [];
        console.log(imgList);
        Array.from(imgList).forEach(elt => {
            let reader = new FileReader();
            console.log(elt);
            reader.onload = (e) => {
                images.push({url: e.target.result, blob: elt});
                this.setState({
                    images: images,
                })
            };
            reader.readAsDataURL(elt);
        });
    }

    render() {
        let renderDiv = null;

        if (this.state.token !== "none") {
            renderDiv = <div id="projectForm">
                <form>
                    <h3>Add image to this project</h3>
                    <div className="form-group">
                        <label htmlFor="images">Images</label>
                        <input className="form-control" type="file" name="images" id="images" accept="image/*" onChange={(e) => this.updateImages(e.target.files)} multiple/>
                        {this.state.images.length > 0 &&
                        this.state.images.map(function (img, i) {
                            return <img key={i} src={img.url} style={{height: 200, width: 'auto'}}/>
                        })
                        }
                    </div>
                    <button type="button" className="btn btn-default" onClick={() => this.onClickUpdate()}>Update</button>
                </form>
            </div>;
        }
        else {
            renderDiv = <div/>;
        }

        return renderDiv;
    }
}

function mapStateToProps(state) {
    const token = state.loginInfos.token;

    return {
        token
    };
}

export default hot(module)(connect(mapStateToProps)(AddImageToProject));