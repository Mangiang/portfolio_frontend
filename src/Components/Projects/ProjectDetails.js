// @flow

import React, {Component} from 'react';
import {hot} from 'react-hot-loader';
import {connect} from 'react-redux';
import {withRouter} from 'react-router';
import ReactHtmlParser from 'react-html-parser';

import {ProgressBar} from '../Other/ProgressBar/ProgressBar';

import type {Dispatch, Project} from '../../Actions/Actions';
import {displayProject} from '../../Actions/Actions';

import {format} from '../../Utilities/DateUtilities';

import style from './styles/ProjectDetails.css'

import Carousel from '../Other/Carousel/Carousel'

type Props = {
    project: Project,
    getProject: (id: string) => Project,
    projectID: string,
}

export class ProjectDetails extends Component<Props> {
    constructor(props: Props) {
        super(props);
    }

    componentWillMount() {
        this.props.getProject(this.props.projectID);
    }

    render() {


        let project = this.props.project;

        return (
            <div key={project.id} className={style['container']}>
                <h1 id={"projectTitle"}>
                    {project.title}
                </h1>
                <p>
                    Starting date : {(project.beginDate ? format(project.beginDate) : "")}
                    <br/>
                    Ending date : {(project.endDate ? format(project.endDate) : "")}
                </p>

                <div className={style["carousel-container"]}>
                    {
                        project.images &&
                        <Carousel slides={project.images} projectTitle={project.title}/>
                    }
                </div>
                <br/>
                <br/>
                <div id={"projectDescription"}
                     className={style['projectDescription']}>{ReactHtmlParser(project.description)}</div>
            </div>
        )
    }
}

function mapStateToProps(state, ownProps) {
    const project = state.projectDetailsReducer;
    const projectID = ownProps.projectID;

    return {
        project: project,
        projectID: projectID
    };
}

function mapDispatchToProps(dispatch: Dispatch) {
    return {
        getProject: (id: string) => {
            dispatch(displayProject(id));
        }
    }
}

export default hot(module)(withRouter(connect(mapStateToProps, mapDispatchToProps)(ProjectDetails)));