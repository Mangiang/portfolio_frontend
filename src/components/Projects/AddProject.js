import React from 'react';
import {hot} from 'react-hot-loader';

import {connect} from 'react-redux';
import {addProject} from '../../actions/actions';


class AddProject extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputTitle: "",
            inputDescription: "",
            beginDate: new Date(),
            endDate: new Date(),
            images:[]
        };

        this.onClickAdd = this.onClickAdd.bind(this);
        this.updateInputDescription = this.updateInputDescription.bind(this);
        this.updateInputTitle = this.updateInputTitle.bind(this);
        this.updateBeginDate = this.updateBeginDate.bind(this);
        this.updateEndDate = this.updateEndDate.bind(this);
        this.updateImages = this.updateImages.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.state.token = nextProps.token;
    }

    componentWillMount() {
        this.state.token = this.props.token;
    }

    onClickAdd() {
        this.props.dispatch(addProject(
            this.state.inputTitle,
            this.state.inputDescription,
            this.state.beginDate,
            this.state.endDate,
            this.state.images,
            this.state.token
        ));
        this.setState({
            inputDescription: "",
            inputTitle: "",
            beginDate: new Date(),
            endDate: new Date(),
            images: []
        })      
    }

    onClickHide() {
        
    }

    updateInputTitle(e) {
        this.setState({
            inputTitle: e.target.value
        })
    }

    updateInputDescription(e) {
        this.setState({
            inputDescription: e.target.value
        })
    }
    
    updateBeginDate(date) {
        this.setState({
            beginDate:date 
        })
    }
    
    updateEndDate(date) {
        this.setState({
            endDate:date 
        })
    }
    
    updateImages(imgList) {
        let imgs = [];
        console.log(imgList);
        Array.from(imgList).forEach(elt => {
            let reader = new FileReader();
            console.log(elt);
            reader.onload = (e) => {
                imgs.push({url:e.target.result, blob:elt});
                this.setState({
                    images:imgs,
                })
            };
            reader.readAsDataURL(elt);
        });
    }

    render() {
        if (this.state.token !== "none")
            return (
                <div id="projectForm">
                    <form>
                        <h3>New project</h3>
                        <div className="form-group">
                            <label htmlFor="titlep">Title</label>
                            <input className="form-control" type="text" name="titlep" id="titlep" onChange={this.updateInputTitle} value={this.state.inputTitle}/>
                        </div>
                        <div className="form-group"> 
                            <label htmlFor="description">Description</label>
                            <input className="form-control" type="description" name="description" id="description" onChange={this.updateInputDescription} value={this.state.inputDescription}/>
                        </div>
                        <div className="form-group"> 
                            <label htmlFor="images">Images</label>
                            <input className="form-control" type="file" name="images" id="images" accept="image/*" onChange={(e) => this.updateImages(e.target.files)} multiple />
                            { this.state.images.length > 0 &&
                                this.state.images.map(function (img, i) {
                                    return <img key={i} src={img.url} style={{height:200, width:'auto'}}/>
                                })
                            }
                        </div>
                        <div className="form-group">
                            <label htmlFor="beginDate">Begining date</label>
                        </div>
                        <div className="form-group">
                            <label htmlFor="endDate">Ending date</label>
                        </div>
                        <button type="button" className="btn btn-default" onClick={() => this.onClickAdd()}>Add</button>
                    </form>                
                </div>
            );
        else
            return <div></div>
    }
}

function mapStateToProps(state) {
    const token = state.loginInfos.token;

    return {
        token
     };
}

export default hot(module)(connect(mapStateToProps)(AddProject));