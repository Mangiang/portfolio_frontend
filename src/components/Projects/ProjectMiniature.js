// @flow

import React from 'react';
import {hot} from 'react-hot-loader';
import {connect} from 'react-redux';
import {withRouter} from 'react-router';
import {Link} from 'react-router-dom';
import styles from './styles/ProjectMiniature.css';
import {StatusToString} from '../../Utilities/ProjectStatus'

type Props = {
    projects: Object,
    projectIdx: number
}

class ProjectMiniature extends React.Component<Props> {
    project: {
        id: string,
        title: string,
        description: string,
        beginDate: Date,
        endDate: Date,
        images: Array<Object>
    };

    constructor(props: Props) {
        super(props);
        this.project = props.projects[props.projectIdx];
        console.log(props.projectIdx, this.project);
    }

    static onClick(id): void {
        console.log("click" + id)
    }

    render() {
        let image = "";
        if (this.project
            && this.project.images !== undefined
            && this.project.images[0] !== undefined) {
            image = <img className={styles['projectMiniatureImage']} src={this.project.images[0][0].url}/>
        }

        return (
            <div className={styles['projectMiniature']}>
                {
                    this.project &&
                    <Link to={'/project/' + this.project.id}>
                        <div className={styles['projectMiniatureLink']}
                             onClick={() => ProjectMiniature.onClick.bind(this, this.project.id)}>
                            {image}
                            <div className={styles['projectName']}>
                                <p className={styles['projectMiniatureTitle']}>{this.project.title}</p>
                                <p className={styles['projectMiniatureBody']}>{this.project.description}</p>
                            </div>
                        </div>
                        <div className={styles['projectMiniatureStatus']}>
                            <div className={styles['projectMiniatureStatusHolder']}>
                                <div className={styles['projectMiniatureStatusIconHolder']}>
                                    <div className={styles['projectMiniatureStatusIconOngoing']}/>
                                </div>
                                <div className={styles['projectMiniatureStatusText']}>{StatusToString(this.project.status)}</div>
                            </div>
                        </div>
                    </Link>
                }
            </div>
        )
    }
}

function mapStateToProps(state, ownProps) {
    const projects = state.projects;
    const projectIdx = ownProps.projectIdx;
    return {
        projects: projects,
        index: projectIdx
    };
}

export default hot(module)(withRouter(connect(mapStateToProps)(ProjectMiniature)));