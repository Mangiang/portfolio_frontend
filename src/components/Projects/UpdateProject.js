import React from 'react';
import {hot} from 'react-hot-loader';

import {connect} from 'react-redux';
import {updateProject} from '../../actions/actions';


class UpdateProject extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputTitle: "",
            inputDescription: "",
            beginDate: new Date(),
            endDate: new Date()
        };

        this.onClickUpdate = this.onClickUpdate.bind(this);
        this.updateInputDescription = this.updateInputDescription.bind(this);
        this.updateInputTitle = this.updateInputTitle.bind(this);
        this.updateBeginDate = this.updateBeginDate.bind(this);
        this.updateEndDate = this.updateEndDate.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.state.token = nextProps.token;
        this.setState({
            inputTitle: nextProps.project.title,
            inputDescription: nextProps.project.description,
            beginDate: new Date(nextProps.project.beginDate),
            endDate: new Date(nextProps.project.endDate)
        })
    }

    componentWillMount() {
        this.state.token = this.props.token;
    }

    onClickUpdate() {
        this.props.dispatch(updateProject({
                id: this.props.id,
                title: this.state.inputTitle,
                description: this.state.inputDescription,
                beginDate: this.state.beginDate,
                endDate: this.state.endDate
            },
            this.state.token
        ));
    }

    updateInputTitle(e) {
        this.setState({
            inputTitle: e.target.value
        })
    }

    updateInputDescription(e) {
        this.setState({
            inputDescription: e.target.value
        })
    }

    updateBeginDate(date) {
        this.setState({
            beginDate: date
        })
    }

    updateEndDate(date) {
        this.setState({
            endDate: date
        })
    }

    render() {
        if (this.state.token !== "none")
            return (
                <div id="projectForm">
                    <form>
                        <h3>Update</h3>
                        <div className="form-group">
                            <label htmlFor="title">Title</label>
                            <input className="form-control" type="text" name="title" id="title" onChange={this.updateInputTitle} value={this.state.inputTitle}/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="description">Description</label>
                            <input className="form-control" type="description" name="description" id="description" onChange={this.updateInputDescription}
                                   value={this.state.inputDescription}/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="beginDate">Begining date</label>
                        </div>
                        <div className="form-group">
                            <label htmlFor="endDate">Ending date</label>
                        </div>
                        <button type="button" className="btn btn-default" onClick={() => this.onClickUpdate()}>Update</button>
                    </form>
                </div>
            );
        else
            return <div/>
    }
}

function mapStateToProps(state) {
    const token = state.loginInfos.token;

    return {
        token
    };
}

export default hot(module)(connect(mapStateToProps)(UpdateProject));