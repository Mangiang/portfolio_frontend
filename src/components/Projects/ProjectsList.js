// @flow

import React, {Component} from 'react';
import {hot} from 'react-hot-loader';

import {connect} from 'react-redux';
import {withRouter} from 'react-router';

import type {Dispatch} from '../../actions/actions';
import {getProjects} from '../../actions/actions';

import ProjectMiniature from './ProjectMiniature'

import styles from './styles/ProjectList.css'

type Props = {
    projects: Array<Object>,
    getProjects: () => any
}

export class ProjectsList extends Component<Props> {
    constructor(props: Props) {
        super(props);
    }

    componentWillMount() {
        this.props.getProjects();
    }

    render() {
        let key = -1;
        return (
            <div className={styles['projectList']}>
                {}
                {
                    this.props.projects.map(() => {
                        key++;
                        return <ProjectMiniature projectIdx={key} key={key}/>;
                    })
                }
            </div>
        )
    }
}

/*

 */

function mapStateToProps(state) {
    const projects = state.projects;

    return {
        projects,
    };
}

function mapDispatchToProps(dispatch: Dispatch) {
    return {
        getProjects: () => {
            dispatch(getProjects());
        }
    }
}

export default hot(module)(withRouter(connect(mapStateToProps, mapDispatchToProps)(ProjectsList)));