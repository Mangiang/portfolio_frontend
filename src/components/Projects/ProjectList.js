import {Component} from 'react';
import {hot} from 'react-hot-loader';

import {connect} from 'react-redux';

import {displayProject} from '../../actions/actions';
import UpdateProject from './UpdateProject';
import AddImageToProject from './AddImageToProject';

import format from 'date-fns/format';

class ProjectList extends Component {
    constructor(props) {
        super(props);
    }

    componentWillMount() {
        this.props.dispatch(displayProject(this.props.id))
    }

    componentWillReceiveProps(nextProps) {

    }

    render() {
        let images = "";
        if (this.props.project.images !== undefined) {
            images = (
                this.props.project.images.map((img, i) => {
                    return (
                        <div key={i} className="image">
                            <a href={img[0].url}>
                                <img src={img[0].url}/>
                            </a>
                        </div>
                    )
                })
            )
        }

        return (
            <div>
                <UpdateProject id={this.props.id} project={this.props.project}/>
                <div id="projectDetails">
                    <h3>{this.props.project.title}</h3>
                    <p>Description : {this.props.project.description}</p>
                    <p>Begin date : {format(this.props.project.beginDate, "DD-MM-YYYY")}</p>
                    <p>End date : {format(this.props.project.endDate, "DD-MM-YYYY")}</p>
                    <div id="projectImages">
                        <AddImageToProject id={this.props.project.id}/>
                        {images}
                    </div>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    const project = state.currentProject;

    return {
        project,
    };
}

export default hot(module)(connect(mapStateToProps)(ProjectList));