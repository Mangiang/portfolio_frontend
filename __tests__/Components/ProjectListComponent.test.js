// @flow

import React from 'react'
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer'
import {ProjectsList} from '../../src/components/Projects/ProjectsList'

describe('>>> PROJECTS LIST COMPONENT --- Snapshot', () => {
    it('+++ capturing Snapshot of Projects list', () => {
        const renderedValue = renderer.create(<ProjectsList projects={[]} getProjects={() => []}/>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('>>> PROJECTS LIST COMPONENT', () => {
    let wrapper;
    const projects = [];
    const getProjects = (): any => [];

    beforeEach(() => {
        wrapper = shallow(<ProjectsList projects={projects} getProjects={getProjects}/>)
    });

    it('+++ render the DUMB component', () => {
        expect(wrapper.length).toEqual(1)
    });
});
