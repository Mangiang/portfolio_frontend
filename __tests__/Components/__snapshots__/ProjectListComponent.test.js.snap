// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> PROJECTS LIST COMPONENT --- Snapshot +++ capturing Snapshot of Projects list 1`] = `<div />`;
